tutel/impls/communicate.py

```diff
@@ -32,6 +32,7 @@ class AllToAllStatus:
     scatter_tensor_shape = None
     num_split = 0
     world_size = 0
+    algo = os.getenv('TUTEL_ALLTOALL_ALGO', '').upper()
```

Comment:
Better adding a parameter in MoELayer than using env.

Reply:
this won't introduce any precision difference for MoE layer, so added it as an env for easy testing, following `NCCL_ALGO` style

@ghostplant any suggestion?

Reply:
For experimental purpose, you can try using environment variable to minimize code changes. But it is better to use `os.environ.get` instead of `os.getenv` here.

Reply:
changed to `os.environ.get`, but these two should be the same?
```
In [1]: import os

In [2]: os.getenv??
Signature: os.getenv(key, default=None)
Source:
def getenv(key, default=None):
    """Get an environment variable, return None if it doesn't exist.
    The optional second argument can specify an alternate default.
    key, default and the result are str."""
    return environ.get(key, default)
```